packages/smooth_app/lib/data_models/user_management_provider.dart

```diff
@@ -82,6 +83,32 @@ class UserManagementProvider with ChangeNotifier {
     notifyListeners();
   }
 
+  /// Check if the user is still logged in and the credentials are still valid
+  /// If not, the user is logged out
+  Future<void> checkUserLoginValidity() async {
+    try {
+      if (ProductQuery.isLoggedIn()) {
+        final User user = ProductQuery.getUser();
+        final bool checkLogin = await OpenFoodAPIClient.login(
+          User(
+            userId: user.userId,
+            password: user.password,
+          ),
+        );
+        if (checkLogin) {
+          // Credentials are still valid so we just return
+          return;
+        } else {
+          // Credentials are not valid anymore so we log out
+          await logout();
+        }
+      }
+    } catch (e) {
+      // We don't want to crash the app if the login check fails
+      // We might want to log this error in the future
+      Logs.e('Login check failed');
```

Comment:
That would mean we get an error each time the user starts the app offline.
I would keep the `catch` but not log the "error", that we can just ignore. Unless we find a better idea.